model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/changes/remote/updater/BaseUpdater.java

```diff
@@ -75,9 +76,17 @@ public final boolean isReadOnly() {
 
     @Override
     public final void markDeleted() {
+        if (isCreated()) {
+            transientEntity = true;
+        }
         state = UpdaterState.DELETED;
```

Comment:
this is in an extreme level of nitpicking, but the boolean needs 8 more bytes in memory. I would suggest to add the value `UpdaterState.TRANSIENT` and change this to
```suggestion
        state = isCreated() ? UpdaterState.TRANSIENT : UpdaterState.DELETED;
```
JOL output with COOPS (compressed references) & CCPS (compressed classes)

```
org.keycloak.models.sessions.infinispan.changes.remote.updater.BaseUpdater object internals:
OFF  SZ           TYPE DESCRIPTION                   VALUE
  0   8                (object header: mark)         N/A
  8   4                (object header: class)        N/A
 12   1        boolean BaseUpdater.transientEntity   N/A
 13   3                (alignment/padding gap)       
 16   8           long BaseUpdater.versionRead       N/A
 24   4              K BaseUpdater.cacheKey          N/A
 28   4              V BaseUpdater.cacheValue        N/A
 32   4   UpdaterState BaseUpdater.state             N/A
 36   4                (object alignment gap)        
Instance size: 40 bytes
Space losses: 3 bytes internal + 4 bytes external = 7 bytes total
```
```
org.keycloak.models.sessions.infinispan.changes.remote.updater.BaseUpdater object internals:
OFF  SZ           TYPE DESCRIPTION               VALUE
  0   8                (object header: mark)     N/A
  8   4                (object header: class)    N/A
 12   4              K BaseUpdater.cacheKey      N/A
 16   8           long BaseUpdater.versionRead   N/A
 24   4              V BaseUpdater.cacheValue    N/A
 28   4   UpdaterState BaseUpdater.state         N/A
Instance size: 32 bytes
```

Reply:
Done!